src/schema/firefox-schemas-import.js

```diff
@@ -504,26 +503,59 @@ inner.isBrowserSchema = (path) => {
   return schemaRegexes.some((re) => re.test(path));
 };
 
-export function fetchSchemas({ inputPath, outputPath, version }) {
-  return new Promise((resolve) => {
-    let tarball;
+/**
+ * Strip a null byte if it's the last character in a string.
+ **/
+export function stripTrailingNullByte(str) {
+  if (str.indexOf('\u0000') === str.length - 1) {
+    return str.slice(0, -1);
+  }
+  return str;
+}
+
+function getTarball({ inputPath, version }) {
+  return new Promise((resolve, reject) => {
     if (inputPath) {
-      tarball = fs.createReadStream(inputPath);
+      resolve(fs.createReadStream(inputPath));
     } else if (version) {
-      tarball = request.get(downloadUrl(version));
+      const url = downloadUrl(version);
+      const tmpPath = path.join('tmp', path.basename(url));
+      // eslint-disable-next-line no-console
+      console.log(`Downloading to ${tmpPath}...`);
+      request
+        .get(url)
+        .pipe(fs.createWriteStream(tmpPath))
+        .on('close', () => {
+          // eslint-disable-next-line no-console
+          console.log('Download completed');
+          resolve(fs.createReadStream(tmpPath));
+        });
+    } else {
+      reject(new Error('inputPath or version is required'));
     }
-    tarball
-      .pipe(zlib.createGunzip())
-      // eslint-disable-next-line new-cap
-      .pipe(tar.Parse())
-      .on('entry', (entry) => {
-        if (inner.isBrowserSchema(entry.path)) {
-          const filePath = path.join(outputPath, path.basename(entry.path));
-          entry.pipe(fs.createWriteStream(filePath));
-        }
-      })
-      .on('end', () => {
-        resolve();
-      });
+  });
+}
+
+export function fetchSchemas({ inputPath, outputPath, version }) {
+  return new Promise((resolve, reject) => {
+    getTarball({ inputPath, version })
+      .then((tarball) => {
+        tarball
+          .pipe(new tar.Parse())
+          .on('entry', (entry) => {
+            // The updated tar library doesn't always strip null bytes from the
+            // end of strings anymore so we get to do that here.
+            const entryPath = stripTrailingNullByte(entry.path);
+            if (inner.isBrowserSchema(entryPath)) {
+              const filePath = path.join(outputPath, path.basename(entryPath));
+              entry.pipe(fs.createWriteStream(filePath));
+            } else {
+              entry.resume();
+            }
+          })
+          .on('end', () => {
+            resolve();
+          });
```

Comment:
Would be nice to take the opportunity to listen to error events and reject there too.